api/src/main/java/jakarta/data/repository/Query.java

```diff
@@ -18,51 +18,77 @@
 package jakarta.data.repository;
 
 import jakarta.data.Sort;
-import jakarta.data.page.CursoredPage;
 import jakarta.data.page.Page;
+import jakarta.data.page.PageRequest;
 
 import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
 /**
- * <p>Annotates a repository method to specify a query string such as SQL, JPQL, Cypher etc. to execute.</p>
- *
- * <p>Jakarta Data providers for relational databases must support
- * JPQL queries if backed by a Jakarta Persistence provider,
- * and otherwise must support SQL queries.</p>
- *
- * <h2>Parameters</h2>
- *
- * <p>The query language can used named parameters or positional parameters.</p>
- *
- * <p><b>Named parameters</b> are referred to by name within the query language.
- * The {@link Param} annotation annotates method parameters to bind them to a named parameter name.
- * The {@code Param} annotation is unnecessary for named parameters when the method parameter name
- * matches the query language named parameter name and the application is compiled with the
- * {@code -parameters} compiler option that makes parameter names available
- * at runtime. When the {@code Param} annotation is not used, the Jakarta Data provider must
- * interpret the query by scanning for the delimiter that is used for positional parameters.
- * If the delimiter appears for another purpose in a query that requires named parameters,
- * it might be necessary for the application to explicitly define the {@code Param} in order to
- * disambiguate.</p>
- *
- * <p><b>Positional parameters</b> are referred to by a number that corresponds to the
- * numerical position, starting with 1, of the repository method parameters.</p>
+ * <p>Annotates a repository method as a query method, specifying a query written in Jakarta Data Query Language (JDQL)
```

Comment:
That might lead to a complex discussion, but using the self-continued principle and using JPQL as a sample, we are once again able to cross-database specification.

```
/**
 * <p>Annotates a repository method as a query method, specifying a query written in Jakarta Data Query Language (JDQL)
 * as the minimum requirement. This encourages the use of JDQL for cross-database compatibility. Additionally, a Jakarta Data
 * provider has the freedom to support extensions to the language, including features from the Jakarta Persistence Query Language (JPQL).
 * While providers are not required to support the complete JPQL language, which primarily targets relational data stores, they are
 * encouraged to provide as broad a support as possible to enhance the flexibility and power of data access queries.</p>
 */
```

Reply:
There's nothing wrong with what you just wrote, just not sure it's quite the right place to be saying things like that.

Reply:
OK I think I understand what you're looking for. Just not quite sure how to say it in a way which isn't out of place in the doc for that annotation. (It's the kind of thing that fits more comfortably in the spec or in `module-info`.

Reply:
@otaviojava please see 628abf4. Is that sorta approximately like what you would like to see?